nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py

```diff
@@ -190,51 +514,96 @@ def __init__(self):
-        # use self.jnt_ctrl.home() for now
-        raise NotImplementedError()
+        # CAUTION: This function should only ever home the arms. Don't add homing of other things here
+        # right gripper is openend on bootup by kinova. not sure where, but not in init
+        if not self.gpr_ctrl.close_right():
+            return False
+        if not self.jnt_ctrl.cmd_left_arm(self.jnt_ctrl.left_arm_home_joint_values):
+            return False
+        home_pose = GeometryPose()
+        # TODO get this from the service
+        home_pose.position.x = 0.581
+        home_pose.position.y = 0.003
+        home_pose.position.z = 0.637
+        home_pose.orientation.x = -0.456
+        home_pose.orientation.y = -0.583
+        home_pose.orientation.z = 0.430
+        home_pose.orientation.w = 0.517
+        if not self.right_cartesian.cmd_orientation(home_pose.orientation):
+            return False
+        if not self.right_cartesian.cmd_position(home_pose.position, True):
+            return False
+        return True
 
-    def extend_handoff(self):
-        # extend right arm in cartesian space
-        # TODO: replace with service call
-        self.jnt_ctrl.cmd_right_arm(CFG["right_arm_extended_handoff"]["joints"])
+    def extend_handoff(self, goal_point: Point):
+        return self.right_cartesian.cmd_position(goal_point, True)
+
+    def extend_restock(self):
+        # TODO: these should come from some param server
+        restock_pose = GeometryPose()
+        restock_pose.position.x = 0.807
+        restock_pose.position.y = 0.053
+        restock_pose.position.z = 0.978
+        return self.right_cartesian.cmd_position(restock_pose.position, True)
 
 
 # test code
 if __name__ == "__main__":
     rospy.init_node("manipulation_control")
     manipulation = ManipulationControl()
-    manipulation.jnt_ctrl.home()
+
+    # close the gripper
+    manipulation.gpr_ctrl.close_right()
+    time.sleep(2)
+    # Move left arm to home with joint ctrl
+    manipulation.jnt_ctrl.cmd_left_arm(manipulation.jnt_ctrl.left_arm_home_joint_values)
+
+    # Move right arm to home in cartesian
+    home_pose = GeometryPose()
+    home_pose.position.x = 0.581
+    home_pose.position.y = 0.003
+    home_pose.position.z = 0.637
+
+    home_pose.orientation.x = -0.456
+    home_pose.orientation.y = -0.583
+    home_pose.orientation.z = 0.430
+    home_pose.orientation.w = 0.517
+
+    handoff_pose = GeometryPose()
+    handoff_pose.position.x = 0.807
+    handoff_pose.position.y = 0.053
+    handoff_pose.position.z = 0.978
+
+    handoff_pose.orientation.x = -0.69435
+    handoff_pose.orientation.y = -0.05705
+    handoff_pose.orientation.z = -0.054539
+    handoff_pose.orientation.w = 0.7153
+
+    rospy.loginfo("Going to home orientation")
+    manipulation.right_cartesian.cmd_orientation(home_pose.orientation)
+    rospy.loginfo("Arrived at home orientation")
+
+    for iter in range(3):
```

Comment:
Should this be here? Not sure if this is going to mess with the testing when the node is brought up

Reply:
this chunk of code only runs if we rosrun this node. during our full runs, we do not rosrun this node. file just defines some classes used elsewhere